Bearded.Utilities/Algorithms/BinPacking.cs

```diff
@@ -107,11 +108,10 @@ internal Result(ReadOnlyCollection<PositionedRectangle<T>> rectangles, int width
         /// <typeparam name="T">Type of custom user data associated with the rectangle.</typeparam>
         /// <param name="rectangles">The rectangles to pack.</param>
         /// <returns>
-        /// Null, if the given list of rectangles is empty.
-        /// Otherwise, an object containing the packed rectangles and some additional information.
+        /// A result object containing the packed rectangles and some additional information.
         /// </returns>
         /// <exception cref="ArgumentNullException"><paramref name="rectangles"/> is null.</exception>
-        public static Result<T> Pack<T>(IEnumerable<Rectangle<T>> rectangles)
+        public static Result<T>? Pack<T>(IEnumerable<Rectangle<T>> rectangles)
```

Comment:
If this algorithm can result null, then I think the documentation should explain what returning null actually means.

Reply:
Right now it's really just a shortcut for not allocating if an empty enumerable is passed in.. Perhaps it would make more sense to return an empty result instead?

Reply:
Agreed